xwayland/server.c

```diff
@@ -332,15 +345,24 @@ static bool server_start(struct wlr_xwayland_server *server) {
 		pid_t pid = fork();
 		if (pid < 0) {
 			wlr_log_errno(WLR_ERROR, "second fork failed");
+			(void) !write(p[1], "", 0);
 			_exit(EXIT_FAILURE);
 		} else if (pid == 0) {
 			exec_xwayland(server);
 		}
 
 		int sig;
 		sigwait(&sigset, &sig);
-		kill(ppid, SIGUSR1);
-		wlr_log(WLR_DEBUG, "sent SIGUSR1 to process %d", ppid);
+		if (write(p[1], "", 0) == -1) {
```

Comment:
Maybe we can just write a single byte like `\n`? I'm worried about a zero-byte write not waking up poll…